hs_core/hydroshare/resource.py

```diff
@@ -478,29 +483,39 @@ def update_resource(
         resource.owners.add(owner)
 
     if edit_users:
+        # TODO: remove edit_users
         resource.edit_users.clear()
         for user in edit_users:
             user = utils.user_from_id(user)
+            # TODO remove edit_users
             resource.edit_users.add(user)
+            # TODO: remove view_users
             resource.view_users.add(user)
 
     if view_users:
+        # TODO: remove view_users
         resource.view_users.clear()
         for user in view_users:
             user = utils.user_from_id(user)
+            # TODO: remove view_users
             resource.view_users.add(user)
 
     if edit_groups:
+        # TODO: remove edit_groups
         resource.edit_groups.clear()
         for group in edit_groups:
             group = utils.group_from_id(group)
             resource.edit_groups.add(group)
+            # TODO: remove view_groups
             resource.view_groups.add(group)
 
+    # FIXME: why does checking view_groups lead to mutating edit_groups & view_groups?
     if view_groups:
+        # TODO: remove edit_groups
         resource.edit_groups.clear()
         for group in view_groups:
             group = utils.group_from_id(group)
+            # TODO: remove view_groups
```

Comment:
I have not got a chance to look through this pull request yet, but want to point out the following:
1. Most changes from Ilan appear to be additions of TODO comments with intention to remove usage of several access-control fields such as owners, edit_users, and view_users. One other change I saw is to add an inactive user into HSAccess DB when a user signs up and before the user's email gets verified. 
2. This pull request is to merge into my updated HSAccess work branch (269-updated-access-control), not to develop branch, so this pull request merge does not affect our develop code.
3. Since the latest decision for HSAccess work is to move HSAccess DB logic into Django DB without introducing a separate DB for easy of filtering and management, this earlier HSACCESS work will most likely not be used at all but only serves as reference. I think moving HSAccess DB logic into Django DB work needs to coordinate with @ironfroggy's resource+metadata model refactoring work since HSAccess model needs to be integrated into the resource model.